src/main/java/com/avaje/ebeaninternal/server/persist/dml/InsertMeta.java

```diff
@@ -18,6 +19,9 @@
  */
 public final class InsertMeta {
 
+  private enum IdGeneration {
+    BY_SERVER, BY_CLIENT, BY_CLIENT_WITH_IDENTIY_INSERT;
```

Comment:
This is the most complex part, basically there are two ways to generate an id
1. by server - when the id property is null (e.g. auto-increment/sequence)
2. by client - when the id property is set (e.g. uuid)

Unfortunately, sqlserver fails if the client tries to set/enforce an id on an auto-increment column. If you want to do this, you must prepend a "SET IDENTITY_INSERT tablename ON" before your insert statement.

So what I do here:
1. Determine if the ID of the bean is generated by the server
2. if yes - Determine, if the platform needs identity_insert
3. if yes - surround the SQL insert with "set identity_insert on/off"